flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -0,0 +1,289 @@
+
+	/**
+	 * Evaluates a SQL query on registered tables and retrieves the result as a {@link Table}.
+	 *
+	 * <p>All tables referenced by the query must be registered in the TableEnvironment.
+	 * A {@link Table} is automatically registered when its {@link Table#toString()} method is
+	 * called, for example when it is embedded into a String.
+	 * Hence, SQL queries can directly reference a {@link Table} as follows:
+	 *
+	 * <pre>
+	 * {@code
+	 *   Table table = ...;
+	 *   String tableName = table.toString();
+	 *   // the table is not registered to the table environment
+	 *   tEnv.sqlQuery("SELECT * FROM tableName");
+	 * }
+	 * </pre>
+	 *
+	 * @param query The SQL query to evaluate.
+	 * @return The result of the query as Table
+	 */
+	Table sqlQuery(String query);
+
+	/**
+	 * Evaluates a SQL statement such as INSERT, UPDATE or DELETE; or a DDL statement;
+	 * NOTE: Currently only SQL INSERT statements are supported.
+	 *
+	 * <p>All tables referenced by the query must be registered in the TableEnvironment.
+	 * A {@link Table} is automatically registered when its {@link Table#toString()} method is
+	 * called, for example when it is embedded into a String.
+	 * Hence, SQL queries can directly reference a {@link Table} as follows:
+	 *
+	 * <pre>
+	 * {@code
+	 *   // register the table sink into which the result is inserted.
+	 *   tEnv.registerTableSink("sinkTable", fieldNames, fieldsTypes, tableSink);
+	 *   Table sourceTable = ...
+	 *   String tableName = sourceTable.toString();
+	 *   // sourceTable is not registered to the table environment
+	 *   tEnv.sqlUpdate(s"INSERT INTO sinkTable SELECT * FROM tableName");
+	 * }
+	 * </pre>
+	 *
+	 * @param stmt The SQL statement to evaluate.
+	 */
+	void sqlUpdate(String stmt);
+
+	/**
+	 * Evaluates a SQL statement such as INSERT, UPDATE or DELETE; or a DDL statement;
+	 * NOTE: Currently only SQL INSERT statements are supported.
+	 *
+	 * <p>All tables referenced by the query must be registered in the TableEnvironment.
+	 * A {@link Table} is automatically registered when its {@link Table#toString()} method is
+	 * called, for example when it is embedded into a String.
+	 * Hence, SQL queries can directly reference a {@link Table} as follows:
+	 *
+	 * <pre>
+	 * {@code
+	 *   // register the table sink into which the result is inserted.
+	 *   tEnv.registerTableSink("sinkTable", fieldNames, fieldsTypes, tableSink);
+	 *   Table sourceTable = ...
+	 *   String tableName = sourceTable.toString();
+	 *   // sourceTable is not registered to the table environment
+	 *   tEnv.sqlUpdate(s"INSERT INTO sinkTable SELECT * FROM tableName", config);
+	 * }
+	 * </pre>
+	 *
+	 * @param stmt The SQL statement to evaluate.
+	 * @param config The {@link QueryConfig} to use.
+	 */
+	void sqlUpdate(String stmt, QueryConfig config);
+
+	/**
+	 * Returns the {@link QueryConfig} depends on the concrete type of this TableEnvironment.
+	 */
+	QueryConfig queryConfig();
```

Comment:
Why do we need this method in the `TableEnvironment` interface? This is not a property of a whole environment, but it is query specific. Therefore I think it does not belong here.

I think it is used as a default value for methods that do not expect `QueryConfig`, but it can be a private internal method, can't it?

Reply:
Good catch. I think we can drop this method in the `TableEnvironment`.